AngularPart1/Components/Home/todo.html

```diff
@@ -0,0 +1,14 @@
+<div>
+    <button class="delete" ng-click="removeTask(task)">x</button>
+    <div ng-click="changeMode(task)">
+        <p><b>Name: </b>
+            <a ng-href="#!{{ task.date }}/edit" class="taskLink">{{ task.name }}</a>
```

Comment:
It is considered a bad practice to generate routes in different parts of application: in case your router config has changed, you will have to go through all of them and change routes again and again. I recommend creating redirect method to one place (dedicated service), it is the simplest way.